File: 283-move-zeroes/move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        
        left_pointer = 0
        l = len(nums)

        for i in range(l):
            if nums[i] != 0:
                nums[left_pointer] = nums[i]
                left_pointer += 1
        
        for i in range(left_pointer,l):
            nums[i] = 0

        return nums

        

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
     int leftPointer=0;
     int i;
     for(i=0;i<nums.length;i++){
        if (nums[i]!=0){
            nums[leftPointer]=nums[i];
            leftPointer++;
        }
     }
     for(i=leftPointer;i<nums.length;i++){
        nums[i] = 0;
     }   
     
    }
}